src/main/java/com/heatheryou/httpserver/RequestParser.java

```diff
@@ -1,17 +1,42 @@
 package com.heatheryou.httpserver;
 
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.util.ArrayList;
 import java.util.List;
 
 public class RequestParser {
+    private List<String> requestList;
 
-    public ResponseBuilder parse(List<String> requestLineList) {
-        String requestString = requestLineList.get(0);
-        String[] requestArray = requestString.split(" ");
-        String method = requestArray[0];
-        String uri = requestArray[1];
-        String protocol = requestArray[2];
-        ResponseBuilder responseLine = new ResponseBuilder(method, uri, protocol);
+    public Request processRequest(BufferedReader requestReader) throws IOException {
+        List<String> requestList = readRequest(requestReader);
+        String[] requestLine = parse(requestList);
+        String uri = getUri(requestLine);
+        String method = getMethod(requestLine);
+        return new Request(uri, method);
+    }
+
+    public List<String> readRequest(BufferedReader requestReader) throws IOException {
```

Comment:
If nothing outside this class actually uses `readRequest` it should probably be a private method. Same for `parse`.

Reply:
There are a few places where I initially made the methods private, then changed them when trying to access them for unit tests. It seems like there are a few ways to get around testing private methods (or not testing them at all), would love to discuss with you on what approach you would suggest I take.

Reply:
Kent Beck in _Test Driven Development_ suggests that while you always write tests to test-drive a method, it's fine to delete them after you've made the method private. With judicious use of git commits, you can always put the tests back if the method is made public again. It's not the only way to handle testing private methods though so feel free to bring up the topic in your apprenticeship channel in Slack and get a range of opinions!